src/lib/Libifl/pbsD_selectj.c

```diff
@@ -97,8 +97,10 @@ __pbs_selectjob(int c, struct attropl *attrib, char *extend)
 		ret = PBSD_select_get(c);
 
 	/* unlock the thread lock and update the thread context data */
-	if (pbs_client_thread_unlock_connection(c) != 0)
+	if (pbs_client_thread_unlock_connection(c) != 0) {
+		free(ret);
```

Comment:
ret is a char **, should we be calling free_string_array() on it?

Reply:
This isn't a NULL ptr terminated string array.  PBSD_select_get() will allocate enough memory for all the strings in one go.  We only need to free it.  It does look weird though.  I'll add a code comment.